homeassistant/components/august/gateway.py

```diff
@@ -0,0 +1,144 @@
+"""Handle August connection setup and authentication."""
+
+import asyncio
+import logging
+
+from august.api import Api
+from august.authenticator import AuthenticationState, Authenticator
+from requests import RequestException, Session
+
+from homeassistant.const import CONF_PASSWORD, CONF_TIMEOUT, CONF_USERNAME
+
+from .const import (
+    CONF_ACCESS_TOKEN_CACHE_FILE,
+    CONF_INSTALL_ID,
+    CONF_LOGIN_METHOD,
+    DEFAULT_AUGUST_CONFIG_FILE,
+    VERIFICATION_CODE_KEY,
+)
+from .exceptions import CannotConnect, InvalidAuth, RequireValidation
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AugustGateway:
+    """Handle the connection to August."""
+
+    def __init__(self, hass):
+        """Init the connection."""
+        try:
+            self._api_http_session = Session()
+        except RequestException as ex:
```

Comment:
How can this happen ? Is there some I/O happening here ?

Reply:
It can't https://github.com/psf/requests/blob/master/requests/sessions.py

Its relocated code that can be reduced away

```
-    try:
-        api_http_session = Session()
-    except RequestException as ex:
-        _LOGGER.warning("Creating HTTP session failed with: %s", str(ex))
```

Reply:
fixed